### PR TITLE
Broaden tensor-type inference coverage with real-world corpus fixtures

### PR DESCRIPTION
**Pull request by khatchad**
Adds six faithful fixtures and their parameter-type tests, each mirroring a decorated function from a real-world TensorFlow project, to broaden tensor-type (shape and dtype) inference coverage. They extend the existing free-function fixtures (`testCrf*`, `testCreateAttentionMaskFromInputMask`, `testMultilayerPerceptron`, …) and `tf.keras.Model.call` method coverage (`testNeuralNetwork*`, `testMultiGPUTraining`, the `autoencoder.py` tests).

Free functions:
- `logistic_regression` — `softmax(W x + b)` over global `tf.Variable`s; param `x` infers as `(100, 784) float32`.
- `nce_loss` — averaged noise-contrastive-estimation loss; params `x_embed` `(4, 10) float32`, `y` `(4, 1) int32`.
- `evaluate` — cosine similarity against an embedding matrix; param `x_embed` `(4, 10) float32`.
- `random_jitter` — random resize/crop/mirror image augmentation; both image params infer as `(256, 256, 3) float32`.

Methods:
- `MaskSparseCategoricalCrossentropy.__call__` — a mask-weighted sparse-categorical-crossentropy loss; all three params infer concretely (`y_true` `(4,) int32`, `y_predict` `(4, 10) float32`, `input_mask` `(4,) float32`).
- `LSTM.call` — a built-in `tf.keras.layers.LSTM` plus a `Dense` read-out; input param `x` infers as `(256, 28, 28) float32`. The forward-pass locals keep their `float32` dtype but report unknown shape (the layer-chain shape gap, wala/ML#530).

Every decorated function has its full input signature recovered. Each fixture runs to completion under `python3.10`, with `assert`s on the runtime shape and dtype matching the static-analysis expectations.

### CODE REVIEW

File: com.ibm.wala.cast.python.test/data/tf2_test_evaluate.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+embedding = tf.Variable(tf.random.normal([vocabulary_size, embedding_size]))
+
+
+def evaluate(x_embed):
+    # Compute the cosine similarity between input data embedding and every embedding vector.
+    x_embed = tf.cast(x_embed, tf.float32)
+    x_embed_norm = x_embed / tf.sqrt(tf.reduce_sum(tf.square(x_embed)))
+    embedding_norm = embedding / tf.sqrt(
+        tf.reduce_sum(tf.square(embedding), 1, keepdims=True), tf.float32
+    )
```
[the remark this answers is not in the document]
Comment (khatchad) [reply com.ibm.wala.cast.python.test/data/tf2_test_evaluate.py:20]: > `embedding_norm` currently calls `tf.sqrt(..., tf.float32)`, but `tf.sqrt`'s second positional argument is `name`, not `dtype`. [...] `x_embed_norm` normalizes by a single scalar norm (no `axis`), which doesn't match the comment about cosine similarity per embedding vector and makes the fixture less faithful.

Both are reproduced verbatim from the analyzed source (`evaluate` in `aymericdamien/TensorFlow-Examples/.../word2vec.py`) — the `tf.sqrt(..., tf.float32)` call (where `tf.float32` lands in the `name` slot) and the scalar `x_embed` normalization are exactly the upstream author's code. These fixtures are deliberately faithful to the real-world source they exercise the analyzer against; "correcting" the logic would make the fixture stop representing the code we actually analyze. It runs to completion under `python3.10`/TF 2.9.3 (the corpus runtime) with the runtime shape/dtype `assert`s passing. I added a code comment flagging the quirk so it reads as intentional rather than a transcription error.
